source/game/system/map/MapdataCheckPath.hh

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include "game/system/map/MapdataAccessorBase.hh"
+
+#include <egg/math/Vector.hh>
+
+namespace System {
+class MapdataCheckPath {
+public:
+    struct SData {
+        u8 start;
+        u8 size;
+
+        u8 prev[6];
+        u8 next[6];
+        u16 _padding;
+    };
+
+    MapdataCheckPath(const SData *data);
+    void read(EGG::Stream &stream);
+
+private:
+    const SData *m_rawData;
+    u8 m_start;               ///< Index of the first checkpoint in this checkpath
+    u8 m_size;                ///< Number of checkpoints in this checkpath
+    std::array<u8, 6> m_prev; ///< Indices of previous connected checkpaths
+    std::array<u8, 6> m_next; ///< Indices of next connected checkpaths
+    s8 m_depth; ///< Number of checkpaths away from the first checkpath (i.e. distance from the
+                ///< start)
```

Comment:
Sorry this is nitpicky, since only one word dangles over to the next line, can you try to shorten slightly just to make formatting look cleaner?
```suggestion
    s8 m_depth; ///< Number of checkpaths away from first checkpath (i.e. distance from start)
```